Add missing chapter warning

GHOSTBEAR opened the pull request:
closes #3732 

## What it does
Its add a warning on the transition page that shows how many chapters is missing if there are any chapters missing else it will not show

## Screenshot
| If chapters are missing | If chapters are not missing | When there are no more chapters |
|-|-|-|
|  ![Screenshot_1599064808](https://user-images.githubusercontent.com/6576096/92011992-260e7300-ed4c-11ea-81d7-4975934b802f.png) | ![Screenshot_1599065385](https://user-images.githubusercontent.com/6576096/92012720-17748b80-ed4d-11ea-90ba-533d7d2c816d.png) | ![Screenshot_1599065594](https://user-images.githubusercontent.com/6576096/92013075-94a00080-ed4d-11ea-8e55-ff82a6a63d68.png) |

arkon commented:
You'll need to tint the icon so it's visible on white/grey backgrounds.

Also it might make sense to put it in between the two chapter names?

GHOSTBEAR commented:
| Gray | White | Webtoon |
|---|---|---|
| ![Screenshot_1599074094](https://user-images.githubusercontent.com/6576096/92026686-a390ae00-ed61-11ea-87a4-c26fb557e8a0.png) | ![Screenshot_1599074109](https://user-images.githubusercontent.com/6576096/92026669-9f649080-ed61-11ea-9e84-d9a7186b98e5.png) | ![Screenshot_1599074123](https://user-images.githubusercontent.com/6576096/92026649-9a9fdc80-ed61-11ea-91ed-c69082c8cd99.png) |

Added tint when ImageView is on white background, gray background seems fine.

Also added the warning between the current/finished and previous/next.

But I feel like the warning message should be different, maybe something more along the lines of `There seems to be X missing chapters. Turn back now if you don't want to be spoiled or if you fine with it go ahead`

arkon commented on app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerTransitionHolder.kt:
```diff
@@ -63,13 +92,43 @@ class PagerTransitionHolder(
+        showMissingChapterWarning(hasMissingChapters)
+    }
+
+    private fun showMissingChapterWarning(boolean: Boolean) {
```
`boolean` -> `visible`

arkon commented on app/src/main/res/values/strings.xml:
```diff
@@ -702,4 +702,8 @@
     <string name="tapping_inverted_both">Both</string>
 
+    <plurals name="missing_chapters_warning">
+        <item quantity="one">Warning there is one missing chapter!</item>
```
Let's keep it short.

"There is 1 missing chapter"

arkon commented on app/src/main/res/values/strings.xml:
```diff
@@ -702,4 +702,8 @@
 
+    <plurals name="missing_chapters_warning">
+        <item quantity="one">Warning there is one missing chapter!</item>
+        <item quantity="other">Warning there are %d missing chapters!</item>
```
"There are %d missing chapters"

arkon commented on app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerTransitionHolder.kt:
```diff
@@ -40,14 +49,34 @@ class PagerTransitionHolder(
         wrapContent()
     }
 
+    private var warningTextView: TextView = TextView(context).apply {
```
Can we make it so that the icon is to the left of the text?

arkon commented on app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerTransitionHolder.kt:
```diff
@@ -40,14 +49,34 @@ class PagerTransitionHolder(
-    private var textView = TextView(context).apply {
-        textSize = 17.5F
+    private var warningImageView: ImageView = ImageView(context).apply {
+        val warningDrawable = resources.getDrawable(R.drawable.ic_warning_white_48dp, resources.newTheme())
+        background = warningDrawable
+        val tintColor = when (preferences.readerTheme().get()) {
+            0 -> Color.BLACK // Theme is White
+            else -> Color.WHITE // Theme is Black or Gray
+        }
```
Use `setVectorCompat` and specify the tint with that.

GHOSTBEAR commented:
| Pager | Webtoon |
|---|---|
| ![Screenshot_1599087407](https://user-images.githubusercontent.com/6576096/92045250-5bcd4f00-ed80-11ea-8e1d-06dc0a106081.png) | ![Screenshot_1599087103](https://user-images.githubusercontent.com/6576096/92045258-5ff96c80-ed80-11ea-97c8-983efe1cd55b.png) |

arkon commented:
Awesome, thanks!
